### PR TITLE
Rearrange order of cops in Rubocop config file

### LINKED ISSUE #1385: Order .rubocop.yml
Order the cops in .rubocop.yml by sections (Style, Rails, etc.) and also alphabetically inside every section. It should look similar to: https://github.com/bbatsov/rubocop/blob/master/config/default.yml (only the order, the cops enabled should remain the same)

Add a *short* description (one sentence) of the cop in a comment in the line before it. Some cops already have it, but there are many without it.

### PR DESCRIPTION
**Pull request by siaw23**
This arranges the sections and subsections in .rubocop.yml in an alphabetical order. Also adds comments explaining each section/subsection.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,39 +1,13 @@
...
-# Use hash literal {} instead of Hash.new
-Style/EmptyLiteral:
-  Enabled: true
-
-# Prefer `each` over `for i`
-Style/For:
-  Enabled: true
-
+AllCops:
+  Include:
```
Comment (Ana06) [inline .rubocop.yml]: Add before this line the following comment: 
`# Include common Ruby source files.`

File: .rubocop.yml [hunk 2]
```diff
@@ -1,39 +1,13 @@
...
-
-# Use hash literal {} instead of Hash.new
-Style/EmptyLiteral:
-  Enabled: true
-
-# Prefer `each` over `for i`
-Style/For:
-  Enabled: true
-
+AllCops:
```
Comment (Ana06) [inline .rubocop.yml]: Add before this line the following comment:
`# Common configuration.`

**Comment by Ana06** [comment]
@siaw23 also please read the [commits](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md#commits) and [pull request](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md#pull-requests-workflow) sections that we recently added to our [contributing guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md). :wink:

**Comment by siaw23** [comment]
ok :) @Ana06

**Comment by siaw23** [comment]
@Ana06 is my commit ok now?

**Comment by differentreality** [comment]
No, not really. You need to reference the issue you are closing on the description of the PR, not its title. Please check our contributing guide again!

File: .rubocop.yml [hunk 3]
```diff
@@ -50,85 +119,37 @@ Lint/LiteralInCondition:
...
-  Include:
-    - '**/Rakefile'
-    - '**/config.ru'
-  Exclude:
-    - 'db/schema.rb'
-    - 'vendor/bundle/**/*'
-    - 'bundle/**/*'
-    - 'config/**/*'
-    - 'bin/*'
+inherit_from: .rubocop_todo.yml
```
Comment (Ana06) [inline .rubocop.yml]: This line should be on top, the first line of this file. As it is importing a file thst overwrite this one,.

**Comment by siaw23** [comment]
how about now? @differentreality @Ana06

File: .rubocop.yml [hunk 4]
```diff
@@ -1,17 +1,41 @@
-Rails:
-  Enabled: true
+# Common configuration.
 inherit_from: .rubocop_todo.yml
```
Comment (Ana06) [inline .rubocop.yml]: this should be on top, before the `Common configuration comment` and line and empty line after it.

**Comment by siaw23** [comment]
@Ana06 now?

**Comment by Ana06** [comment]
yes! now it is great!

But there are merge conflicts after merging https://github.com/openSUSE/osem/pull/1378

Please rebase :wink:

**Comment by siaw23** [comment]
are we good now? :P @Ana06

**Comment by Ana06** [comment]
@siaw23 you have to squash your commits, so all the changes appear in the first commit. :wink:

**Comment by siaw23** [comment]
i only see one commit in my logs @Ana06

**Comment by Ana06** [comment]
There are two: 23dab1a2e6b6dc50e48b0ba8b96d3d67ff7b81ad

How did you rebase?

**Comment by differentreality** [comment]
Perhaps you haven't pushed after you rebased? We need to be able to see your changes here in github as well, check the tab on the top of the page saying 'Commits', it still shows 2 commits.

**Comment by siaw23** [comment]
hmmm this is new to me. i see the 2 commits locally, by when i try to start the process to squash, i don't see the last commit that i want to squash 🤔😳

`Rearrange the order of cops` is my commit. I can't see the one i rebased from (the merged one)

```
pick f78f449 Added demo data
pick db2d776 Sign in Ui improved in mobile
pick 8e4cca9 removed the render banners component test
pick 3ef17e9 Removed the View Tests
pick 2c33384 Improve contributing guide
pick 123c9f6 Add mock_auth config for Github
pick 3655c80 Fixed registration limit issue
pick 43dbb15 fix: #1345 (remove duplicate text in error message)
pick 9c2a09e changed dropdown for small devices
pick d561718 Rearrange the order of cops
pick e8699f4 Updating readme, install, payments, contributing files
pick e5aa6ab Fix bug in User#registered_to_event?
pick 9d4c02c Improve test coverage for User model
pick 09696a4 fixes bug OSEM#1298
pick 29e77db Fix link to TRANSLATION guide in readme
pick 1ca448d Changed Ruby syntax in views. Fixed #1326
pick 5e9bc68 Move reports to separate controller
pick 7de009e Enable Style/DoubleNegation Rubocop cop
pick dc41daa added feature test for roles

# Rebase f7f5ab8..23dab1a onto f7f5ab8 (19 commands)
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
# d, drop = remove commit
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
```

**Comment by Ana06** [comment]
what command did you execute to rebase?

make a copy of your changes (as it is easy to delete your changes when rebasing) and then run:

`git remote -v`

your should see something like:

```
origin  https://github.com/Ana06/osem.git (fetch)
origin  https://github.com/Ana06/osem.git (push)
upstream        https://github.com/openSUSE/osem.git (fetch)
upstream        https://github.com/openSUSE/osem.git (push)
```

in case that upstream doesn't appear also add:

`git remote add upstream https://github.com/openSUSE/osem.git` 

and then when running `git remote -v` it should appear


And then rebase to master:

`git rebase upstream master`

And the extra commit should have disappear. In case that you have problem or need more help fell free to ping me in IRC (@Ana06) or ask in the #osem channel on freenode IRC

**Comment by Ana06** [comment]
@siaw23 do you plan to reopen this? I didn't merge this other PR that is already ready to merge: https://github.com/openSUSE/osem/pull/1398 to merge yours first, so just tell me if you will finish it soon, otherwise I'll merge the other Pr and you'll have to rebase again. 

Let's me know if you need help to reopen it :wink:

**Comment by siaw23** [comment]
yeah i'm doing it. just threw some questions in IRC to clarify something.

**Pull request closed without merging**